projects/fhi-angular-components/src/lib/fhi-tree-view-selection/fhi-tree-view-selection.component.ts

```diff
@@ -93,39 +94,55 @@ export class FhiTreeViewSelectionComponent implements OnInit, OnChanges {
       if (item.children && item.children.length > 0) {
         this.updateCheckedState(id, item.children, multiToggle, checkAll);
       }
-      item.descendantStateConfirmed = false;
     });
   }
 
-  private updateDecendantState(items: Item[], initialState: boolean) {
-    items.forEach((item) => {
-      if (item.children && item.children.length > 0) {
-        if (item.children.every((item) => item.descendantStateConfirmed)) {
-          if (item.children.find((item) => item.isChecked) !== undefined) {
-            this.setHasCheckedDescendantAndIsExpanded(true, item, initialState);
-          } else {
-            this.setHasCheckedDescendantAndIsExpanded(false, item, initialState);
-          }
-          if (item.children.find((item) => item.hasCheckedDescendant) !== undefined) {
-            this.setHasCheckedDescendantAndIsExpanded(true, item, initialState);
-          }
-          item.descendantStateConfirmed = true;
+  private updateDecendantState(
+    items: Item[],
+    expandCheckedItems: boolean,
+  ): FhiTreeViewSelectionItemState {
+    const itemsState = {
```

Comment:
Should this be typed?

```suggestion
    const itemsState: FhiTreeViewSelectionItemState = {
```